src/renderer/views/Changes/subcomponents/CommitForm/CommitForm.tsx

```diff
@@ -0,0 +1,129 @@
+import { Box, Typography } from '@mui/material';
+import { useFormik } from 'formik';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch, useSelector } from 'react-redux';
+import { selectRepoTree } from '../../../../../shared/redux/slices/repoStatusSlice';
+import * as checkStatus from '../../../../../shared/utils/repoStatus/statusChecks';
+import * as repoTree from '../../../../../shared/utils/repoStatus/tree';
+import Section from '../../../../components/Section/Section';
+import TextField from '../../../../components/TextField/TextField';
+import { validationSchema, FormFields } from './validationSchema';
+import * as Styled from './style';
+import Button from '../../../../components/Button/Button';
+import TextArea from '../../../../components/TextArea/TextArea';
+import { gitCommit } from '../../../../ipc';
+import FilesByFolder from '../ChangedFiles/FilesByFolder';
+import { sendNotification } from '../../../../../shared/redux/slices/notificationsSlice';
+
+interface Props {
+  closeForm: () => void;
+}
+
+const CommitForm: React.FC<Props> = ({ closeForm }) => {
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const formik = useFormik<FormFields>({
+    initialValues: {
+      summary: '',
+      description: '',
+    },
+    validationSchema,
+    onSubmit: ({ summary, description }, { setSubmitting }) => {
+      gitCommit(`${summary}\n\n${description}`);
+      setSubmitting(false);
+      closeForm();
+      dispatch(
+        sendNotification({
+          title: t('Notification.success'),
+          message: t('Changes.prompts.success'),
+          type: 'success',
+          autoDismiss: true, // if not set the default is false
+        })
+      );
+    },
+  });
+  const tree = useSelector(selectRepoTree);
+  const commited = tree
+    ? repoTree.search(
+        tree,
+        (node) =>
+          checkStatus.isChanged(node.status) &&
+          checkStatus.isStaged(node.status)
+      )
+    : [];
```

Comment:
@LostInStatic do you think this would potentially benefit from a `useMemo`?

Reply:
IMO not really - there are two cases when the `commited` array could not be recomputed (`closeForm` prop is ignored, it should not change without parent rerendering):
 - when the parent component rerenders: It would require a different setup than just wrapping it in `useMemo`. Also, for now in all the cases (I can think of) the component should be unmounted anyway.
 - when `tree` is updated, but stays the same: AFAIK it would require a deep comparison, which would be as resource-intensive as the recomputation.